### PR TITLE
Update number inputs

### LINKED ISSUE #1141: Implement consistent number formatting for fields with numbers
We have lots of fields (tables) that require an input of numbers. We'd like this to be a consistent experience, which formats using commas, dollar signs, and percentage signs. Dollars need to be rounded to the nearest dollar, and we need to be able to accommodate 14 characters ($1,000,000,000). Referencing #1049 for where this work originated.

47188125-94c6bc80-d2eb-11e8-99ce-800cab059602

### PR DESCRIPTION
**Pull request by mgwalker**
Closes #1141

### This pull request changes...
- adds commas to dollar inputs; these are automatic - manual commas are still silently deleted
  screen shot 2018-11-19 at 2 36 20 pm

- allows percent inputs to include 2 digits after the decimal point
  screen shot 2018-11-19 at 2 36 02 pm

### This pull request is ready to merge when...
- [x] Tests have been updated (and all tests are passing)
- [x] This code has been reviewed by someone other than the original author

### This feature is done when...
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

**Comment by mgwalker** [comment]
I'd probably advocate punting on doing checks on the percentages without doing validation everywhere else.  We had #831 but it got closed (presumably it's layer 2 or higher?)

**Comment by lauraponce** [comment]
That makes sense, it's not a super high priority, more of a nice to have to minimize errors I would say. Should I create a new issue for that and tag it with future enhancement so we don't lose sight of this down the line, and then I could sign off on this PR?

File: web/src/stories/index.js [hunk 1]
```diff
@@ -32,3 +32,30 @@ storiesOf('Buttons', module).add('various', () => (
...
+      console.log(e.target.value);
+      this.setState({ value: e.target.value });
+    };
+
+    render() {
+      return (
+        <DollarInput
+          hideLabel
+          wrapperClass="m0"
+          className="fake-spacer-input m0 input input-condensed mono right-align"
```
Comment (hbillings) [inline web/src/stories/index.js]: Oh, shoot, I forgot to remove the `fake-spacer-input` from the classlist when I merged the table PR. There isn't an associated class, so you can drop it from here.

File: web/src/stories/index.js [hunk 2]
```diff
@@ -32,3 +32,30 @@ storiesOf('Buttons', module).add('various', () => (
     <Btn kind="outline" /> <Btn kind="outline" size="big" />
   </div>
 ));
+
+storiesOf('Numeric inputs', module).add('Dollars', () => {
+  class Dollar extends Component {
+    state = { value: 0 };
```
Comment (hbillings) [inline web/src/stories/index.js]: I'm a little confused about this pattern. I'm used to seeing component state initialized in a constructor -- can you help me understand why this is different?

**Comment by NAretakis** [comment]
There’s probably a good amount of data validation that we’ll want to have, but it’s not Layer 1. A future enhancement issue for this would be good, and we can figure out what layer to bake it into in the future.

File: web/src/stories/index.js (hunk 2 above)
Comment (mgwalker) [reply web/src/stories/index.js]: Oops, I didn't even mean to include this.  😆 I just threw it in so I could modify and test them faster than reloading the whole app.  But maybe it's worth keeping, in which case I'll clean it up.

But to your question, it's just less code.  AFAIK it works the same, it's just more declarative.  Putting the state declaration in the constructor is one extra hoop you have to jump through to understand the structure of the class.  It's a [stage-3 proposal](https://github.com/tc39/proposal-private-methods) so it should make it into Javascript officially in the near future.

**Comment by NAretakis** [comment]
@mgwalker looking at the build in a few sections, I realized that having two decimal places for percentages seems weird (since everything else is rounded to nearest whole). Can we remove the decimals, and just use whole numbers, i.e. "78%"?

**Comment by mgwalker** [comment]
Yep, we can do that.  Back to the old way!

**Comment by NAretakis** [comment]
Sorry for the rework ☹

**Pull request merged**